algorithms-miscellaneous-1/src/test/java/com/baeldung/algorithms/factorial/FactorialUnitTest.java

```diff
@@ -0,0 +1,72 @@
+package com.baeldung.algorithms.factorial;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigInteger;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class FactorialUnitTest {
+
+    Factorial factorial;
+
+    @Before
+    public void setup() {
+        factorial = new Factorial();
+    }
+
+    @Test
+    public void givenNumber_whenCallingFactorialUsingForLoop_calcualeFactorial() {
```

Comment:
The test names are missing the "then" component, and for most of them, the "given" isn't really necessary. This one, for example, could be "whenCalculateFactorialUsingForLoop_thenCorrect"